online-shop/src/app/app-routing.module.ts

```diff
@@ -1,19 +1,20 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 import { ProductListComponent } from './product-list/product-list.component';
-import {ProductDetailsComponent} from "./product-details/product-details.component";
-import {HeaderComponent} from "./header/header.component";
-import {AppComponent} from "./app.component";
+import { ProductDetailsComponent } from './product-details/product-details.component';
+import { HeaderComponent } from './header/header.component';
+import {ShoppingCartComponent} from "./shopping-cart/shopping-cart.component";
 
 const routes: Routes = [
-  {path: '', redirectTo: 'home', pathMatch: 'full'},
-  {path: 'home', component: HeaderComponent},
-  {path: 'products', component: ProductListComponent},
-  {path:'products/product/:id', component: ProductDetailsComponent}
-  ];
+  { path: '', redirectTo: 'home', pathMatch: 'full' },
+  { path: 'home', component: HeaderComponent },
+  { path: 'products', component: ProductListComponent },
+  { path: 'product/:id', component: ProductDetailsComponent },
+  { path: 'cart', component: ShoppingCartComponent}
```

Comment:
routes are matched from top to bottom, you should also add a wildcard route -> '**' it will match any route that wasn't caught, if you change the URL and instead of cart you write carts or catrs you won't get any matches because you don't have those routes. so you should get navigated to home or somewhere else, as you like